app/models/maintenance_stats/queries/github/releases_query.rb

```diff
@@ -0,0 +1,81 @@
+module MaintenanceStats
+  module Queries
+    module Github
+      class RepoReleasesQuery < BaseQuery
+        RELEASES_QUERY = Rails.application.config.graphql.client.parse <<-GRAPHQL
+          query($owner: String!, $repo_name: String!, $cursor: String){
+            repository(owner: $owner, name: $repo_name){
+              releases(first: 100, after: $cursor, orderBy: {field: CREATED_AT, direction: DESC}){
+                nodes {
+                  name
+                  publishedAt
+                }
+                totalCount
+                pageInfo {
+                  hasPreviousPage
+                  hasNextPage
+                  endCursor
+                  startCursor
+                }
+              }
+            }
+          }
+        GRAPHQL
+
+        VALID_PARAMS = [:owner, :repo_name, :end_date]
+        REQUIRED_PARAMS = [:owner, :repo_name]
+
+        def self.client_type
+          :v4
+        end
+
+        def query(params: {})
+          validate_params(params)
+
+          end_date = params[:end_date]
+          params = params.slice!(:end_date)
+
+          @releases = nil
+
+          page_info = { has_next_page: true, cursor: nil}
+          while(page_info[:has_next_page])
+            params[:cursor] = page_info[:cursor] if page_info[:cursor]
+            result = @client.query(RELEASES_QUERY, variables: params)
+
+            if check_for_errors(result)
+              @releases = result if @releases.nil?
+              return @releases
+            end
+
+            has_next_page = result.data.repository.releases.page_info.has_next_page
+            cursor = result.data.repository.releases.page_info.end_cursor
+            page_info = { has_next_page: has_next_page, cursor: cursor}
+
+            if @releases.nil?
+              @releases = result
+              next
+            end
+            
+            result.data.repository&.releases.nodes.each do |release|
+              publish_date = DateTime.parse(release.published_at)
+              if publish_date > end_date
+                @releases.data.repository.releases.nodes << release
+              else
+                has_next_page = false
```

Comment:
Should this be modifying `page_info`?

Reply:
`page_info` is a local variable in this loop and reinitialized after each api request.